clusterDeployer.py

```diff
@@ -380,6 +379,9 @@ def cb() -> None:
         self.update_dnsmasq()
 
     def create_workers(self) -> None:
+        if len(self._cc.workers) == 0:
+            logger.info("No worker to setup. Skip")
+            return
         logger.info("Setting up workers")
```

Comment:
move this back out. The function is called `create_workers`, not `create_workers_if_need`.

Ideally, we would write the code so that 0 workers are treated as 1 or more workers to keep the code the same